lib/bowling.rb

```diff
@@ -0,0 +1,110 @@
+
+class Bowling
+
+  attr_reader :frames, :frame, :spare_counter, :strike_counter, :strike_counter2
+  
+  def initialize
+    @frames = []
+    @frame = []
+    @spare_counter = []
+    @strike_counter = []
+    @strike_counter2 = []
+  end
+  
+  def roll(pins)
```

Comment:
Okay lol so my first comment would be that you could split each of these if statements into their own method, preferably with a name describing what exactly it does?

I can see that you were really thorough about accounting for every possibility, but as an outsider it's quite hard to work out what each of the if statements actually means. So you could split each of them into their own method with a name like strike_followed_by_spare etc and it would be a lot easier to follow and also be more in line with SRP.

I also think that it would be a good idea to totally separate rolls and calculating bonuses. When I first started doing it I also had them together but it just ended up getting so complicated... so maybe have the rolls be input, and then have calculating bonuses as something totally separate based on the rolls/frames you have. Just an idea. I think that will help you cut down on all the different if/elsif!